packages/application-generic/src/services/in-memory-provider/in-memory-provider.service.ts

```diff
@@ -37,13 +38,18 @@ export class InMemoryProviderService {
   public inMemoryProviderClient: InMemoryProviderClient;
   public inMemoryProviderConfig: InMemoryProviderConfig;
 
-  constructor(private enableAutoPipelining?: boolean) {
+  constructor(
+    private getFeatureFlag: GetFeatureFlag,
```

Comment:
This aches my brain for 2 different reason: I am plugging a use case in a service to avoid having to build a middle man service `NovuFeatureFlagsService` and also because the pattern we chose for instantiation forces me to have to inject in every use of `InMemoryProviderService` when this dependency could be self contained and instantiated inside of the constructor saving hassles. 🤷🏻

Reply:
First I feel like the `GetFeatureFlag` should be a service, not a use-case, because it exposes many other public functions through its API, and it's against the idea of use-case.
Second, you can also pass the `isClusterMode` flag to the constructor.

Reply:
The second suggestion would probably disappear that need with the forthcoming refactor InMemoryProviderService needs.

About the first I agree and disagree at the very same time.
I do agree that `getWithContext` and `getGlobal` methods from the use case should live in the FeatureFlagsService, but still don't solve the problem.
The use case is where the business rules should live in the domain, that in this case are the default and fallback values of every feature flag plus the key they are related. If I converted this into a service we would expose that information and that's something I consider we shouldn't do it.
I think the problem that I am in would need another layer that we still haven't defined in our architecture. Also I don't recall us having this situation before of 2 interdependent services.
Should we bring this situation to the tech summit to debate how we want to face it as a standard?

Reply:
`The use case is where the business rules should live in the domain` - the same you can say about the service, but the differentiation is that the use-case is more granular, and it encapsulates "one business rule, one use-case".

`If I converted this into a service we would expose that information and that's something I consider we shouldn't do it.` - what about right now, isn't it already exposed?

```
await this.getFeatureFlag.isMultiProviderConfigurationEnabled(...)
```

I think that we can move the `getWithContext` and `getGlobal` to the `FeatureFlagsService` as you mentioned. The rest functions should be defined as separate use-cases or in a service that will expose them.

Reply:
> `If I converted this into a service we would expose that information and that's something I consider we shouldn't do it.` - what about right now, isn't it already exposed?
> 
> ```
> await this.getFeatureFlag.isMultiProviderConfigurationEnabled(...)
> ```

What I meant with expose the information is to have in an infrastructure service like FeatureFlagsService to have the domain information of each feature flag default and fallback value. And take the risk of that in some way to be spread around the apps when implementing their usage (like someone deciding to have to pass a new fallback or default) because that would lead to inconsistencies.

If the main problem is that I called use case to a group of individual feature flags implementations inside one class, I can split them in multiple classes with the command pattern implementation we use. But it disappoints me that this implementation was ok until few days before the release giving me less margin to work in them.

I have other disagreements with some of the architecture related statements of your message but this is not the place to discuss them. Worth for an architecture discussion in a tech summit better. 🙂